src/electionguard_gui/components/upload_ballots_component.py

```diff
@@ -25,23 +25,39 @@ def expose(self) -> None:
         eel.expose(self.upload_ballot)
 
     def create_ballot_upload(
-        self, election_id: str, device_file_name: str, device_file_contents: str
+        self,
+        election_id: str,
+        device_file_name: str,
+        device_file_contents: str,
+        ballot_count: int,
     ) -> dict[str, Any]:
         try:
             db = self._db_service.get_db()
             self._log.debug(f"creating upload for {election_id}")
             election = self._election_service.get(db, election_id)
             if election is None:
                 return eel_fail(f"Election {election_id} not found")
+            created_at = datetime.utcnow()
             ballot_upload_id = self._ballot_upload_service.create(
-                db, election_id, device_file_name, device_file_contents
+                db,
+                election_id,
+                device_file_name,
+                device_file_contents,
+                ballot_count,
+                created_at,
+            )
+            self._election_service.append_ballot_upload(
```

Comment:
Duplicated ballot info into the election so it could easily get displayed on the election page